environs/bootstrap/bootstrap.go

```diff
@@ -350,9 +350,25 @@ func bootstrapImageMetadata(
 		Arches:    availableTools.Arches(),
 		Stream:    environ.Config().ImageStream(),
 	})
-	publicImageMetadata, _, err := imagemetadata.Fetch(sources, imageConstraint)
-	if err != nil {
-		return nil, errors.Annotate(err, "searching image metadata")
+	logger.Debugf("constraints for image metadata lookup %v", imageConstraint)
+
+	// Get image metadata from all data sources.
+	// Since order of data source matters, order of image metadata matters too. Append is important here.
+	var publicImageMetadata []*imagemetadata.ImageMetadata
+	for _, source := range sources {
+		sourceMetadata, _, err := imagemetadata.Fetch([]simplestreams.DataSource{source}, imageConstraint)
+		if err != nil {
+			logger.Debugf("ignoring image metadata in %s because %v", source.Description(), err)
```

Comment:
suggestion:
    s/ because/:/

otherwise you might end up with something like
    "ignoring image metadata in some-source because fail: something went wrong"
which reads weirdly. whereas
    "ignoring image metadata in some-source: fail: something went wrong"
is more in line with the usual formatting